### PR TITLE
OCaml 5.3 support

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

**Comment by pmetzger** [comment]
Looks at worst harmless in that it only impacts 5.3 and after. However, I think as part of this we should extend the CI to 5.3 since the CI didn't actually check that.

File: src/lib/uTop_compat.ml [hunk 1]
```diff
@@ -67,7 +67,28 @@ let rec is_persistent_path = function
 #endif
 
 let invalid_package_error_to_string err =
-#if OCAML_VERSION >= (5, 2, 0)
+#if OCAML_VERSION >= (5,3,0)
```
Comment (kit-ty-kate) [inline src/lib/uTop_compat.ml:70]: this section can be nested in the original one. Only a two lines are actually different

Comment (Octachron) [reply src/lib/uTop_compat.ml:70]: I would suggest  to define `inline_code` compatibility helper rather than deduplicate the full printer.

Comment (anmonteiro) [reply src/lib/uTop_compat.ml:70]: I thought only the newest cppo had nesting support, so I avoided doing it. Am I wrong?

**Comment by anmonteiro** [comment]
Thanks for the helpful code review. The patch is way more minimal now.

**Comment by pmetzger** [comment]
Okay, so what's next?

**Comment by anmonteiro** [comment]
I'm happy to lend some time if utop needs more maintainers

**Review by Octachron — APPROVED** [review]
The changes look good to me.

**Comment by anmonteiro** [comment]
thanks for merging. would be nice to put together an opam release of utop compatible with OCaml 5.3

**Comment by Octachron** [comment]
I will cut a release after the OCaml 5.3.0~beta1 and the 5.2.1 release (aka at the end of this week or next week).

**Pull request merged**